db/schema.rb

```diff
@@ -0,0 +1,62 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `bin/rails
+# db:schema:load`. When creating a new database, `bin/rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema[7.0].define(version: 2022_04_19_224729) do
```

Comment:
- [ ] As the previous reviewer requested, kindly change your associations, table names, and column names, in order to match the ERD guidelines.


## Explanation

Apologies for the inconsistency of previous reviewers who overlooked this requirement on your peers' projects.  This is from the project requirements:

![image](https://user-images.githubusercontent.com/57421823/164382461-7a37e5a6-832f-4f8f-bc20-3c91e0c0695b.png)

The _**only**_ name that needs to be changed from this diagram is the "ENTITY" table, which in this case seems to be your "EXPENSE" table (good).

All the other table and column names and associations, are part of the requirement and are not optional.

_Note: In order to write this answer to your inquiry, this case was internally investigated_

## Hints

- In order for the task to be easier, please note that the class name (model) could be used for associations ie. `belongs_to :author, class_name: 'User'`, that way you can keep your changes to a minimum.
- This Stack Overflow [answer](https://stackoverflow.com/questions/471416/how-do-you-write-a-migration-to-rename-an-activerecord-model-and-its-table-in-ra) covers a way to change the table names, the side effects, and the solutions (look at the answers' comments).
- This Stack Overflow [answer](https://stackoverflow.com/questions/1992019/how-can-i-rename-a-database-column-in-a-ruby-on-rails-migration) covers the way to change the column names on a given table, also the side effects, and the solutions for those side effects.
- After correcting the table and column names, you need to update the models.  Remember: by using this trick  `belongs_to :author, class_name: 'User' you can preserve most of the model names (if not all).
- If you had to change a model name, don't forget to update the routes, controllers, and views.

After making these changes, the ERD will meet the requirements, and your project will be closer to approval 💪

Reply:
Thanks @Meltrust for putting things in a proper context. As for @Mre55, I'm available to assist if you need help, thank you.

Reply:
Hi @elisha2kyakpo1,

I'll be glad if you assist me if you have time.

Reply:
@Mre55, Search for my name on slack and just send me a meeting link.